### ISSUE TITLE
数据集

### ISSUE THREAD

**Issue by windy-hy** [bm2-lab/CausCell#2]
您好，我正在尝试复试论文中相关实验数据，但代码中貌似没有做真实数据集的拆分，可以方便提供一下吗

**Comment by GaoYiChengTJ** [comment]
Hi, we adopted the five-fold cross validation for the split of dataset in this study.

**Comment by windy-hy** [comment]
Thanks for your reply!When reproducing, I found an issue: there is no "Treatment" label in the ICI_Response dataset. Is it replaced by "TreatmenTime"?

**Comment by windy-hy** [comment]
Another issue: when I use your demo code, I find that the accuracy achieved is much higher if I directly call model.disentanglement for disentanglement on the test set without going through model.data_transformation processing.Why is that?

```
testing_data_pwd ="../Data/Merfish_Brain_testing_data.h5ad"
model.load_trained(concept_list=concept_list, concept_counts=concept_counts, concept_cdag=concept_cdag,
                    results_folder=results_folder,
                    trained_profile_size=374,
                    milestone=50)
concept_embs = model.disentanglement(testing_data_pwd=testing_data_pwd,
                                        saved_pwd=results_folder,
                                        concept_list=concept_list,
                                     concept_counts=concept_counts,
                                     concept_cdag=concept_cdag)
model.concept_prediction(testing_data_pwd, concept_embs, concept_list, concept_counts, concept_cdag)

```
when using testing_data_pwd = "../Data/transformed_Merfish_Brain_testing_data.h5ad"
```
Factor_Age: ACC:0.6120151183608514	Precision:0.5849821109028253	Recall:0.6120151183608514	F1:0.5728093292861203
Factor_Domain: ACC:0.4886811219415158	Precision:0.4331667020806148	Recall:0.4886811219415158	F1:0.45339409234713335
Factor_Celltype: ACC:0.3117167296598369	Precision:0.2849278457864246	Recall:0.3117167296598369	F1:0.29746495440792153
```
when using testing_data_pwd = "../Data/Merfish_Brain_testing_data.h5ad"
```
Factor_Age: ACC:0.6122140441615278	Precision:0.5852354247883058	Recall:0.6122140441615278	F1:0.5729942459430889
Factor_Domain: ACC:0.5797095683310125	Precision:0.5379644950835173	Recall:0.5797095683310125	F1:0.550666968440086
Factor_Celltype: ACC:0.9146210463497115	Precision:0.8536050594075489	Recall:0.9146210463497115	F1:0.8796691872969493
```

**Comment by windy-hy** [comment]
Hi, I have already solved this problem. When processing the test set, we should use the mapping relationship that was applied when processing the training set; however, this aspect is not reflected in the demonstration code.
```
    def data_transformation(self, 
                            data_pwd, 
                            save_pwd, 
                            concept_list,
                            log_norm=False,
                            reuse_mapping=False):
        data = sc.read_h5ad(data_pwd)
        if log_norm:
            normed_data = data.X / data.X.sum(axis=1)[:,None] * 10000
            exp_data = np.log(normed_data + 1)
        else:
            exp_data = data.X
        new_obs = []
        label_categories = []
        for idx, factor_name in enumerate(concept_list):
            # if test set
            if reuse_mapping:
                val2idx = joblib.load(f"{save_pwd}/{factor_name}_dict.pkl")
            # if training set
            else:
                factor_vals = list(data.obs[factor_name].unique())
                val2idx = dict(zip(factor_vals, range(len(factor_vals))))
                joblib.dump(val2idx, f"{save_pwd}/{factor_name}_dict.pkl")
            new_obs.append([val2idx.get(i, -1) for i in data.obs[factor_name]])
            label_categories.append(len(val2idx))
        new_df = pd.DataFrame(list(zip(*new_obs)), columns = concept_list)
        new_data = ad.AnnData(exp_data)
        new_data.obs = new_df
        data_name = data_pwd.split("/")[-1]
        new_data.write(f"{save_pwd}/transformed_{data_name}")
        return new_data,label_categories
```
```
transformed_test_data = model.data_transformation(data_pwd="../Data/Merfish_Brain_testing_data.h5ad",
                                                save_pwd="../Data",
                                                concept_list=concept_list,
                                                log_norm=False,
                                                reuse_mapping=True)
```
